bindings/nodejs/tests/closestSoletta.js

```diff
@@ -0,0 +1,19 @@
+var fs = require( "fs" );
+var path = require( "path" );
+var async = require( "async" );
+var dirToCheck;
+
+// Find the closest instance of soletta via its package.json
+function closestSoletta( startDir ) {
+	for ( dirToCheck = startDir; dirToCheck !== "/";
+			dirToCheck = path.resolve( dirToCheck, ".." ) ) {
+		try {
+			if ( JSON.parse( fs.readFileSync(
+					path.join( dirToCheck, "package.json" ), "utf8" ) ).name == "soletta" ) {
+				return dirToCheck;
+			}
+		} catch( errors ) {}
```

Comment:
Similarly, if ```JSON.parse()``` throws an exception, that's also very much expected because we do not accept a malformed ```package.json``` and we must keep moving up the tree looking for the one we want.

Ultimately, if soletta's package.json is not found in the tree, the tests will definitely fail - which is their job :)